### PR TITLE
feat(swc/plugin): wasmer@3 

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

require https://github.com/swc-project/swc/pull/5507 to rebase.

**BREAKING CHANGE:**

**Related issue (if exists):**
 - Closes https://github.com/swc-project/swc/issues/5480

### CODE REVIEW

File: crates/swc_plugin_runner/src/memory_interop.rs [hunk 1]
```diff
@@ -1,58 +1,48 @@
...
-    // is linear and we have atomic guarantee by not having any thread access,
-    // so can safely get subarray from allocated ptr address.
-    //
-    // If we want safer operation instead, refer previous implementation
-    // https://github.com/swc-project/swc/blob/1ef8f3749b6454eb7d40a36a5f9366137fa97928/crates/swc_plugin_runner/src/lib.rs#L56-L61
-    unsafe {
-        view.subarray(ptr_start_size, ptr_start_size + serialized_len_size)
-            .copy_from(serialized_bytes.as_slice());
-    }
+    println!(
```
Comment (kdy1) [inline crates/swc_plugin_runner/src/memory_interop.rs:37]: Please remove this line

File: crates/swc_plugin_runner/src/memory_interop.rs [hunk 2]
```diff
@@ -1,58 +1,48 @@
...
+        "-----------write_into_memory_view {} {}",
+        ptr_start, ptr_start_size
+    );
+
+    // Note: it's important to get a view from memory _after_ alloc completes
+    let view = memory.view(store);
+    let x = view.write(ptr_start_size, serialized_bytes.as_slice());
+    println!("{:#?}", x);
```
Comment (kdy1) [inline crates/swc_plugin_runner/src/memory_interop.rs:45]: Also this

File: crates/swc_plugin_runner/src/memory_interop.rs (hunk 1 above)
Comment (kwonoj) [reply crates/swc_plugin_runner/src/memory_interop.rs:37]: :ohno:

**Review by kdy1 — APPROVED** [review]
Thanks!

---
swc-bump:
 - ''

**Review by kdy1 — DISMISSED** [review]

**Review by kdy1 — APPROVED** [review]

**Pull request merged**